Source/WebKit/UIProcess/ios/forms/WKDatePickerPopoverController.mm

```diff
@@ -96,16 +93,42 @@ - (instancetype)initWithDatePicker:(UIDatePicker *)datePicker
         [[_backgroundView trailingAnchor] constraintEqualToAnchor:self.trailingAnchor],
         [[_backgroundView topAnchor] constraintEqualToAnchor:self.topAnchor],
         [[_backgroundView bottomAnchor] constraintEqualToAnchor:self.bottomAnchor constant:-toolbarBottomMargin],
-        [[_datePicker heightAnchor] constraintEqualToConstant:datePickerSize.height],
-        [[_datePicker leadingAnchor] constraintEqualToAnchor:self.leadingAnchor],
-        [[_datePicker trailingAnchor] constraintEqualToAnchor:self.trailingAnchor],
-        [[_datePicker topAnchor] constraintEqualToAnchor:self.topAnchor],
-        [[_datePicker bottomAnchor] constraintEqualToSystemSpacingBelowAnchor:[_accessoryView topAnchor] multiplier:1],
+        [[mainView heightAnchor] constraintEqualToConstant:mainViewSize.height],
+        [[mainView leadingAnchor] constraintEqualToAnchor:self.leadingAnchor],
+        [[mainView trailingAnchor] constraintEqualToAnchor:self.trailingAnchor],
+        [[mainView topAnchor] constraintEqualToAnchor:self.topAnchor],
+        [[mainView bottomAnchor] constraintEqualToSystemSpacingBelowAnchor:[_accessoryView topAnchor] multiplier:1],
         [[_accessoryView leadingAnchor] constraintEqualToAnchor:self.leadingAnchor constant:accessoryViewHorizontalMargin],
         [[_accessoryView trailingAnchor] constraintEqualToAnchor:self.trailingAnchor constant:-accessoryViewHorizontalMargin],
         [[_accessoryView heightAnchor] constraintEqualToConstant:accessoryViewSize.height],
         [[_accessoryView bottomAnchor] constraintEqualToAnchor:[_backgroundView bottomAnchor]],
     ]];
+}
+
+#if HAVE(UICALENDARSELECTIONWEEKOFYEAR)
+- (instancetype)initWithCalendarView:(UICalendarView *)calendarView selectionWeekOfYear:(UICalendarSelectionWeekOfYear*)weekSelection
+{
+    if (!(self = [super initWithFrame:CGRectZero]))
+        return nil;
+
+    _calendarView = calendarView;
+    [_calendarView setCalendar:[NSCalendar calendarWithIdentifier:NSCalendarIdentifierISO8601]];
+    [_calendarView setSelectionBehavior:weekSelection];
+
+    [self setupView:calendarView toolbarBottomMargin:2];
```

Comment:
I think I had left an earlier comment about pulling out the `2` into a constant at the top of the file, since it's used in two places.